linera-core/src/client.rs

```diff
@@ -591,30 +594,12 @@ where
                 break;
             }
             if self.options.message_policy.is_reject() {
-                if message.event.is_skippable() {
+                if message.bundle.is_skippable() {
                     continue;
-                } else if message.event.is_tracked() {
+                } else if message.bundle.is_tracked() {
                     message.action = MessageAction::Reject;
                 }
             }
-            if let Message::System(SystemMessage::RegisterApplications { applications }) =
```

Comment:
Are these no longer needed because of the blobs API?

Reply:
Once `ApplicationDescription`s are stored as blobs, the `RegisterApplications` message will be removed, yes. But for now, we still need it.

This line here was an optimization to filter out redundant `RegisterApplications` messages and skip them. Now that we have to make the decision whether to skip per bundle rather than per message, we can't skip individual messages anymore, so we can't do this optimization anymore. There's no need to find an alternative, though, since the message is going to be removed anyway.